movie.js

```
@@ -0,0 +1,228 @@
+const searchSubmit = document.querySelector(".search-area-submit");
+const searchText = document.querySelector(".search-area-text");
+const searchResultList = document.querySelector(".search-result-list");
+const pagination = document.querySelector(".pagination");
+const localStorage = window.localStorage;
+const favMovieObj = JSON.parse(localStorage.getItem("favList")) || [];
+const myFavorite = document.createElement("button");
+const body = document.querySelector("body");
+const preview=document.querySelector(".preview");
+let localData;
+let totalSearchResults = 0;
+let n= 0;
+if (JSON.parse(localStorage.getItem("favList"))!==null){
+  n=JSON.parse(localStorage.getItem("favList")).length
+}
+//this is to insert search results using dom
+function searchResult(body) {
+  body.Search.forEach(movie => {
+    let searchResultContainer = document.createElement("li");
+    searchResultContainer.className = "movie";
+    searchResultContainer.id = movie.imdbID;
+    let searchResultTitle = document.createElement("h2");
+    let searchResultYear = document.createElement("h3");
+    let posterImage = document.createElement("img");
+
+    searchResultTitle.innerHTML = `${movie.Title}`;
+    searchResultTitle.className = "movie-title";
+    searchResultContainer.appendChild(searchResultTitle);
+
+    searchResultYear.innerHTML = `${movie.Year}`;
+    searchResultYear.className = "movie-year";
+    searchResultContainer.appendChild(searchResultYear);
+
+    posterImage.src = movie.Poster;
+    posterImage.className = "movie-image";
+    posterImage.alt = "Movie poster";
+    searchResultContainer.appendChild(posterImage);
+
+    let favorite = document.createElement("button");
+    favorite.innerHTML = "Add to favorites";
+    favorite.addEventListener("click", function(event) {
```

It would be better to use event delegation. Let the events bubble up and handle them in one place rather than place place event handlers on elements individually